datafusion/core/tests/memory_limit/mod.rs

```diff
@@ -313,11 +313,9 @@ async fn sort_spill_reservation() {
         // the sort will fail while trying to merge
         .with_sort_spill_reservation_bytes(1024);
 
+    // TODO since we col->row conversion first, sort succeeds without limited sort_spill_reservation_bytes
     test.clone()
-        .with_expected_errors(vec![
-            "Resources exhausted: Additional allocation failed with top memory consumers (across reservations) as:",
-            "B for ExternalSorterMerge",
-        ])
+        .with_expected_success()
```

Comment:
I'm wondering whether the `sort_spill_reservation_bytes` configuration will still be necessary after this change. Since we're now estimating the memory required for the merge phase based on the actual ratio measured from the first batch, the manual override may become less important in practice.